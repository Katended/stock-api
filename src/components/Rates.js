import React, { useEffect } from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';
import { useDispatch, useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
// import Card from 'react-bootstrap/Card';
import { fetchCurrencyRates } from '../redux/home/currencyRatesSlice';
import RatesItem from './RatesItem';
import './Home.css';

const Rates = () => {
  const { id } = useParams();
  const dispatch = useDispatch();
  const rates = useSelector((state) => state.currencyRatesStore.rates);

  useEffect(() => {
    console.log(rates.length === 0);
    if (rates.length === 0) {
      dispatch(fetchCurrencyRates(id));
    }
  }, [rates]);

  return (
    <div className="cases-by-symbol">

      <span className="topheading">
        {id }
      </span>
      <div className="search-by-symbol">
        <div className="filter-inputs">
          <span>&nbsp;</span>
          <span>CURRENCY SYMBOLS</span>
        </div>
      </div>
      <div className="flex-container-details">

        <div>
          {' '}
          <span>
            DATE:
            {' '}
            {(rates.length > 0) ? rates[0].date : ' ' }
          </span>
          {' '}
          <span>
            BASE:
            {' '}
            {id}
          </span>
          {' '}
          EXCHANGE RATES
        </div>
        {
      rates.map((key) => (
        <RatesItem key={key.base} id={key.base} value={key.value} />
      ))
    }
      </div>
    </div>
  );
};

export default Rates;
