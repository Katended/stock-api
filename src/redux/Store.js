import { configureStore } from '@reduxjs/toolkit';
import homeReducer from './home/homeSlice';
import currencyRatesReducer from './home/currencyRatesSlice';

const Store = configureStore({
  reducer: {
    homeStore: homeReducer,
    currencyRatesStore: currencyRatesReducer,
  },
});

export default Store;
