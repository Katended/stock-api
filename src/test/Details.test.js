import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import Store from '../redux/Store';
import Rates from '../components/Rates';

describe('Rates component', () => {
  it('renders Rates correctly', () => {
    const tree = render(<Provider store={Store}><Router><Rates /></Router></Provider>);
    expect(tree).toMatchSnapshot();
  });
});
