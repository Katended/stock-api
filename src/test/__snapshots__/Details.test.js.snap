// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rates component renders Rates correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="cases-by-symbol"
      >
        <span
          class="topheading"
        />
        <div
          class="search-by-symbol"
        >
          <div
            class="filter-inputs"
          >
            <span>
               
            </span>
            <span>
              CURRENCY SYMBOLS
            </span>
          </div>
        </div>
        <div
          class="flex-container-details"
        >
          <div>
             
            <span>
              DATE:
               
               
            </span>
             
            <span>
              BASE:
               
            </span>
             
            EXCHANGE RATES
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="cases-by-symbol"
    >
      <span
        class="topheading"
      />
      <div
        class="search-by-symbol"
      >
        <div
          class="filter-inputs"
        >
          <span>
             
          </span>
          <span>
            CURRENCY SYMBOLS
          </span>
        </div>
      </div>
      <div
        class="flex-container-details"
      >
        <div>
           
          <span>
            DATE:
             
             
          </span>
           
          <span>
            BASE:
             
          </span>
           
          EXCHANGE RATES
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
